src/protocol/graphics.ts

```diff
@@ -238,20 +238,28 @@ export function setupGraphics(store: UIStore) {
     const pause = ThreadFront.currentPause;
     assert(pause);
 
-    const rv = await pause.repaintGraphics();
-    if (pause === ThreadFront.currentPause && rv) {
-      const { mouse } = await getGraphicsAtTime(ThreadFront.currentTime);
-      let { description, screenShot } = rv;
-      if (screenShot) {
-        repaintedScreenshots.set(description.hash, screenShot);
-      } else {
-        screenShot = repaintedScreenshots.get(description.hash);
-        if (!screenShot) {
-          console.error("Missing repainted screenshot", description);
-          return;
+    try {
+      const rv = await pause.repaintGraphics();
+      if (pause === ThreadFront.currentPause && rv) {
+        const { mouse } = await getGraphicsAtTime(ThreadFront.currentTime);
+        let { description, screenShot } = rv;
+        if (screenShot) {
+          repaintedScreenshots.set(description.hash, screenShot);
+        } else {
+          screenShot = repaintedScreenshots.get(description.hash);
+          if (!screenShot) {
+            console.error("Missing repainted screenshot", description);
+            return;
+          }
         }
+        paintGraphics(screenShot, mouse);
+      }
+    } catch (e) {
+      if (e.code == 33) {
+        const { screen, mouse } = await getGraphicsAtTime(ThreadFront.currentTime);
+        paintGraphics(screen, mouse);
```

Comment:
It isn't always valid to paint these graphics at this point.  If we await on the repaint, the user navigates somewhere else, and then the repaint throws, we can't update the graphics.  This is what the "if (pause === ...)" block above is for.